dev/tests/eval-tests.nix

```diff
@@ -2,59 +2,257 @@
 #
 #     nix build -f dev checks.x86_64-linux.eval-tests
 
-rec {
-  f-p = builtins.getFlake (toString ../..);
-  flake-parts = f-p;
+let
+  flake-parts = builtins.getFlake (toString ../..);
+  lib = flake-parts.inputs.nixpkgs-lib.lib;
+in
+(lib.makeExtensibleWithCustomName "extendEvalTests" (evalTests: {
+  inherit evalTests;
+  inherit flake-parts;
+  flake-parts-lib = evalTests.flake-parts.lib;
+  inherit lib;
 
   devFlake = builtins.getFlake (toString ../.);
-  nixpkgs = devFlake.inputs.nixpkgs;
+  nixpkgs = evalTests.devFlake.inputs.nixpkgs;
 
-  f-p-lib = f-p.lib;
+  inherit (evalTests.flake-parts-lib) mkFlake;
+  weakEvalTests.callFlake = { ... } @ flake:
+    let
+      sourceInfo = flake.sourceInfo or { };
+      inputs = flake.inputs or { };
+      outputs = flake.outputs inputs;
+      result = outputs;
+    in
+    result;
+  strongEvalTests.callFlake = { ... } @ flake:
+    let
+      sourceInfo = { outPath = "/unknown_eval-tests_flake"; } //
+        flake.sourceInfo or { };
+      inputs = flake.inputs or { };
+      outputs = flake.outputs (inputs // { self = result; });
+      result = outputs // sourceInfo // {
+        inherit inputs outputs sourceInfo;
+        _type = "flake";
+      };
+    in
+    assert builtins.isFunction flake.outputs;
+    result;
```

Comment:
Do we really need `weakEvalTests.callFlake`?

Not happy about how complicated this got. Relying on laziness and overriding means that it's not immediately obvious what runs how.